### ISSUE TITLE
grad_norm:nan

### ISSUE THREAD

**Issue by YiJiangYue** [exiawsh/StreamPETR#8]
hello, thanks for your work~
when i train using "stream_petr_r50_flash_704_bs1_8key_2grad_24e.py", grad_norm is nan.
Is this normal?

**Comment by exiawsh** [comment]
Hi, the grad norm exists nan or inf issues, but it doesn't affect the performance.
Besides, I do not recommend using stream_petr_r50_flash_704_bs1_8key_2grad_24e.py. The training time is too long. Try to use stream training instead and increase training epochs.

**Comment by YiJiangYue** [comment]
thanks for your reply~ 
I have another question, when i use my own data and close focal head, grad_norm is always NAN and  loss suddenly becomes 0 at some epoch because network weights become NAN, do you have any suggestions for this problem?
thanks again~

**Comment by exiawsh** [comment]
"Grad_norm is always NAN and loss suddenly becomes 0 at some epoch" means that the network does not converge.
There are some details that need to be confirmed:
1. Have you used query denoising during training?
2. Can the single-frame version of PETR converge on your dataset?
3. Try to reduce the learning rate and increase the warmup iterations.
4. How many object queries have you used and what is the detection range （point_cloud_range in config）?

**Comment by exiawsh** [comment]
You may add my WeChat: kira1027293308. 
We can discuss the details.

**Comment by exiawsh** [comment]
We have closed this issue because the problem has been resolved in wechat.
The conclusion is：
Removing focal head may lead to non convergence issue in some datasets.
